backend/metadata_writer/src/metadata_writer.py

```diff
@@ -137,13 +137,22 @@ def is_kfp_v2_pod(pod) -> bool:
 
 while True:
     print("Start watching Kubernetes Pods created by Argo")
-    for event in k8s_watch.stream(
-        k8s_api.list_namespaced_pod,
-        namespace=namespace_to_watch,
-        label_selector=ARGO_WORKFLOW_LABEL_KEY,
-        timeout_seconds=1800,  # Sometimes watch gets stuck
-        _request_timeout=2000,  # Sometimes HTTP GET gets stuck
-    ):
+    if namespace_to_watch:
+        pod_stream = k8s_watch.stream(
```

Comment:
I think in python scope of this var is now limited to this if branch, you need to declare

pod_stream = None

before the if command

Can you verify?